video/out/mac/title_bar.swift

```diff
@@ -91,34 +91,54 @@ class TitleBar: NSVisualEffectView {
     }
 
-    func set(appearance: Any) {
-        if appearance is Int {
-            window?.appearance = appearanceFrom(string: String(appearance as? Int ?? 0))
-        } else {
-            window?.appearance = appearanceFrom(string: appearance as? String ?? "auto")
-        }
+    func set(appearance: Int32) {
+        window?.appearance = { switch appearance {
+            case 1: return NSAppearance(named: .aqua)
+            case 2: return NSAppearance(named: .darkAqua)
+            case 3: return NSAppearance(named: .vibrantLight)
+            case 4: return NSAppearance(named: .vibrantDark)
+            case 5: return NSAppearance(named: .accessibilityHighContrastAqua)
+            case 6: return NSAppearance(named: .accessibilityHighContrastDarkAqua)
+            case 7: return NSAppearance(named: .accessibilityHighContrastVibrantLight)
+            case 8: return NSAppearance(named: .accessibilityHighContrastVibrantDark)
+            case 0: return nil
+            default: return nil
+            }
+        }()
     }
 
-    func set(material: Any) {
-        if material is Int {
-            self.material = materialFrom(string: String(material as? Int ?? 0))
-        } else {
-            self.material = materialFrom(string: material as? String ?? "titlebar")
-        }
+    func set(material: Int32) {
+        self.material = { switch material {
+            case 0: return .titlebar
+            case 1: return .selection
+            case 2: return .menu
+            case 3: return .popover
+            case 4: return .sidebar
+            case 5: return .headerView
+            case 6: return .sheet
+            case 7: return .windowBackground
+            case 8: return .hudWindow
+            case 9: return .fullScreenUI
+            case 10: return .toolTip
+            case 11: return .contentBackground
+            case 12: return .underWindowBackground
+            case 13: return .underPageBackground
+            case 14: return .dark
+            case 15: return .light
+            case 16: return .mediumLight
+            case 17: return .ultraDark
+            default: return .titlebar
+            }
+        }()
     }
 
-    func set(color: Any) {
-        if color is String {
-            layer?.backgroundColor = NSColor(hex: color as? String ?? "#00000000").cgColor
-        } else {
-            let col = color as? m_color ?? m_color(r: 0, g: 0, b: 0, a: 0)
-            let red   = CGFloat(col.r)/255
-            let green = CGFloat(col.g)/255
-            let blue  = CGFloat(col.b)/255
-            let alpha = CGFloat(col.a)/255
-            layer?.backgroundColor = NSColor(calibratedRed: red, green: green,
-                                             blue: blue, alpha: alpha).cgColor
-        }
+    func set(color: m_color) {
+        let red   = CGFloat(color.r)/255
+        let green = CGFloat(color.g)/255
+        let blue  = CGFloat(color.b)/255
+        let alpha = CGFloat(color.a)/255
```

Comment:
Maybe we would like some syntax sugar? 
```
let components = [color.r, color.g, color.b, color.a].map { CGFloat($0) / 255 }
```

Reply:
i get the sentiment behind this, though it makes the NSColor init call a bit more verbose when accessing the components array with the corresponding indices.

maybe it would just be better to not assign the values of the components to any vars and pass them directly to the constructor:
```
        layer?.backgroundColor = NSColor(calibratedRed: CGFloat(color.r)/255,
                                         green: CGFloat(color.g)/255,
                                         blue: CGFloat(color.b)/255,
                                         alpha: CGFloat(color.a)/255).cgColor
```